src/fmu/tools/rms/generate_petro_jobs_for_field_update.py

```diff
@@ -0,0 +1,506 @@
+            geomodel grid into the ERTBOX grid used for field parameter update in ERT.
+            Export the petrophysical properties for each of the facies to ERT in
+            ROFF format. Use a script that take as input the facies realization and
+            the petrophysical properties per facies and use the facies realization
+            as filter to select which values to copy from the petrophysical realizations
+            to put into the petrophysical property parameter that was generated by
+            the original job. This will overwrite the petrophysical properties
+            in the realization from the original job for those
+            parameters that are used as field parameters to be updated by ERT.
+            The other petrophysical parameters generated by the original job but not
+            used as field parameters in ERT will be untouched.
+
+        When updated ensemble is fetched from ERT (ERT iteration > 0):
+
+            Run the original petrosim job to generate all petrophysical parameters
+            in the geogrid. Import the updated field parameters for petrophysical
+            properties per facies into ERTBOX grid.
+            Copy the petrophysical property parameters that were updated as
+            field parameters by ERT into geomodel grid from the ERTBOX grid.
+            This operation is the same as the last step
+            when initial ensemble realization was created. This step will then
+            overwrite the petrophysical property parameters already created by
+            the original job by the new values updated by ERT. All petrophysical
+            property parameters not updated as field parameters by ERT will be
+            untouched and will therefore have the same values as they
+            had after the original petrosim job was run.
+
+
+    Summary
+
+        The current function is made to simplify the preparation step of the
+        RMS project by creating the necessary petrosim jobs for a workflow
+        supporting simultaneous update of both facies and petrophysical
+        properties in ERT.
+
+    Usage of this function
+
+        Input
+            Name of config file with specification of which field parameters
+            to simulate per zone per facies.
+
+        Optional parameters
+            debug info (True/False),
+            report field parameters not used (True/False)
+
+        Output
+            A set of new petro sim jobs to appear in RMS project.
+
+    """
+
+    spec_dict = read_specification_file(config_file)
+    create_new_petro_job_per_facies(
+        spec_dict, debug_print=debug, report_unused=report_unused
+    )
+
+
+@no_type_check
+def check_rms_project(project):
+    if not isinstance(project, _roxar.Project):  # type: ignore
+        raise RuntimeError("This run must be ran in an RoxAPI environment!")
+
+
+def read_specification_file(config_file_name, check=True):
```

Comment:
It would in general be beneficial that you have used [type-annotations](https://runestone.academy/ns/books/published/fopp/Functions/TypeAnnotations.html) in all functions, e.g. a in [example here](https://github.com/equinor/fmu-tools/blob/3b001d3991cbc33bf9cc9c909ccfe47f8d3690d3/src/fmu/tools/domainconversion/dconvert.py#L127): It will make it easier to read code and also to track errors.

Reply:
Fixed